source/common/filesystem/win32/filesystem_impl.cc

```diff
@@ -0,0 +1,100 @@
+#include <fcntl.h>
+#include <io.h>
+#include <sys/stat.h>
+#include <windows.h>
+
+#undef DELETE
+#undef GetMessage
```

Comment:
I'm sure there's a good reason why this was necessary, can you add a comment above, or a link to info if it's a common technique for win32?

Reply:
This is common for getting envoy to compile on Windows:
https://github.com/envoyproxy/envoy/blob/caf7ab123964cedd172a2d4cb29b2f2e05ca9156/source/common/filesystem/win32/directory_iterator_impl.h#L5-L9

Reply:
Thanks for that super detailed explanation. Can you either add a short comment here pointing to directory_iterator_impl.h comment, or replicate that comment here?

Reply:
Done.